### DISCUSSION TITLE
ArkUI: Add more documentation on Combobox

### DISCUSSION THREAD

**Discussion by xeinebiu** [chakra-ui/ark#2954] [Q&A] [answered]
I was trying the combobox and somehow the callback that provides back the selected value was returning only the first item no matter which item you click.

After like 1 hour of debuggning I found out that the Collection class we create, we should implement the value callback and provide the ID of the Item.

It works fine if you provide a collection of strings or a collection that has a Value property but if that is not the case, the value callback of the Collection must be implemented.

Please add on the docs an explaination that this must be implemented.

**Comment by cschroeter** [comment, marked as answer]
https://ark-ui.com/react/docs/guides/collection

**Reply by xeinebiu** [reply]
Thanks @cschroeter!

Wouldn't it make more sense for "value" to match the type we pass to the collection, instead of using a collection of strings?

To me, this solution seems overengineered. 

```jsx
collection={students}
disabled={isLoading}
value={[student.id]}  // Why not just use 'student'? The combobox already holds the collection and can retrieve the id with the itemToValue: (item) => item.id. The collection is already generic.
```

**Reply by cschroeter** [reply]
Great question. I will bring this up in the next chat with @segunadebayo 

For now you can work with items using the `onValueChange` call back as shown below:
<img width="933" alt="image" src="https://github.com/user-attachments/assets/db1b6c38-7881-4a47-8cfa-ebdf847e0b96">

**Comment by Rangoons** [comment]
Similar to this, if we are using a combobox with `allowCustomValue: true`, it looks like we should be updating the collection to contain that value. Which event should we be using to do this? Both `onInputValueChange` & `onValueChange` don't seem to have the value when enter is pressed.
